redis/src/cluster_routing.rs

```diff
@@ -902,25 +894,15 @@ pub enum SlotAddr {
 /// which stores only the master and [optional] replica
 /// to avoid the need to choose a replica each time
 /// a command is executed
-#[derive(Debug, Eq, PartialEq)]
-pub(crate) struct SlotAddrs {
-    pub(crate) primary: String,
-    pub(crate) replicas: Vec<String>,
-}
-
-impl SlotAddrs {
-    pub(crate) fn new(primary: String, replicas: Vec<String>) -> Self {
-        Self { primary, replicas }
-    }
-
-    pub(crate) fn from_slot(slot: Slot) -> Self {
-        SlotAddrs::new(slot.master, slot.replicas)
-    }
+#[derive(Debug, Eq, PartialEq, Clone, PartialOrd, Ord)]
+pub(crate) struct ShardAddrs {
```

Comment:
Does it make sense to wrap the entire struct with `Arc` rather than each property inside it?

Reply:
We need the node addresses (e.g. struct properties) to be wrapped with Arcs to avoid extra copies since the addresses are being retrieved for command executions. For example see addresses_for_all_primaries, slot_addr_for_route